### ISSUE TITLE
Not found page when successfully login in another browser tab

### ISSUE THREAD

**Issue by ElenaLapshyna** [keycloak/keycloak#34965]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

authentication

### Describe the bug

When I open 2 browser tabs with login pages for different clients and successfully login into one tab, the other tab reloads and displays the error "Page not found" (URL became {base_url}/realms/{realm}/protocol/openid-connect/{uuid}).

https://github.com/user-attachments/assets/155b6d35-239c-43b9-82ab-644fe36c56fd

### Version

26.0.4

### Regression

- [ ] The issue is a regression

### Expected behavior

When I open 2 browser tabs with login pages for different clients and successfully login into one tab, the other tab reloads and displays the home page for logged-in users.

### Actual behavior

When I open 2 browser tabs with login pages for different clients and successfully login into one tab, the other tab reloads and displays the error "Page not found" (URL became {base_url}/realms/{realm}/protocol/openid-connect/{uuid}). See the attached video in the "Describe the bug" section.

### How to Reproduce?

1. Open 1st browser tab with login page for `client_1`.
2. Open 2nd browser tab with login page for `client_2` (same browser).
3. Log in user in tab 1.
4. Navigate to tab 2. You see that tab 2 was reloaded and displayed a "Not found page" error.

### Anything else?

I'm using the keycloak image `quay.io/phasetwo/phasetwo-keycloak:26.0.4`.
The start command is: 
```
"command": [
                "start",
                "--db",
                "postgres",
                "--db-url-host",
                "{value}",
                "--db-username",
                "{value}",
                "--db-password",
                "{value}",
                "--spi-email-template-provider=freemarker-plus-mustache",
                "--spi-email-template-freemarker-plus-mustache-enabled=true",
                "--spi-theme-cache-themes",
                "false",
                "--log-level=INFO"
            ],
```
Environment variables are:
```
            "environment": [
                {
                    "name": "KC_PROXY_HEADERS",
                    "value": "forwarded"
                },
                {
                    "name": "KC_DB_PASSWORD",
                    "value": "{value}"
                },
                {
                    "name": "KEYCLOAK_ADMIN_PASSWORD",
                    "value": "{value}"
                },
                {
                    "name": "JGROUPS_DISCOVERY_PROTOCOL",
                    "value": "JDBC_PING"
                },
                {
                    "name": "KC_DB_USERNAME",
                    "value": "{value}"
                },
                {
                    "name": "KEYCLOAK_ADMIN",
                    "value": "{value}"
                },
                {
                    "name": "KC_DB_URL_HOST",
                    "value": "{value}"
                },
                {
                    "name": "KC_HTTP_ENABLED",
                    "value": "true"
                },
                {
                    "name": "KC_DB_URL_DATABASE",
                    "value": "{value}"
                },
                {
                    "name": "KC_DB_URL_PORT",
                    "value": "{value}"
                },
                {
                    "name": "KC_FEATURES",
                    "value": "token-exchange,update-email"
                },
                {
                    "name": "SENTRY_RELEASE",
                    "value": "staging"
                },
                {
                    "name": "KC_HEALTH_ENABLED",
                    "value": "true"
                },
                {
                    "name": "KC_HOSTNAME_BACKCHANNEL_DYNAMIC",
                    "value": "true"
                },
                {
                    "name": "KC_METRICS_ENABLED",
                    "value": "true"
                },
                {
                    "name": "SENTRY_DSN",
                    "value": "{value}"
                },
                {
                    "name": "PROXY_ADDRESS_FORWARDING",
                    "value": "true"
                },
                {
                    "name": "KC_CACHE_CONFIG_FILE",
                    "value": "cache-ispn-jdbc-ping.xml"
                },
                {
                    "name": "KC_HOSTNAME_STRICT",
                    "value": "false"
                },
                {
                    "name": "SENTRY_ERRORS_ONLY",
                    "value": "true"
                },
                {
                    "name": "KC_HOSTNAME",
                    "value": "https://{host_value}"
                }
            ],
```
We discover this behavior after updating the Keycloak version from 24.0.5 to 26.04. Nothing in the logs related to 404 error. 
I can assume, that it can be related to proxy settings. Could you tell me please, what exactly could be the reason and what should I change to fix it? Thank you!

**Comment by mposolda** [comment]
@ElenaLapshyna Can you clarify if you see this behaviour just for some clients or for all clients? For example can you try with the Keycloak built-in client `account` and open the account login page in 2 browser tabs and then check that after login in tab1, what would be the behaviour of tab1?

Otherwise do you have any idea what is the `{uuid}` at the bottom of the URL and what this UUID refers to? I mean the UUID like `de6df442-...` which you saw in your example. Does it refer to `client_uuid` of the target client? If not, does this value differ for every client?

**Comment by ElenaLapshyna** [comment]
@mposolda thank you for your response. 
This behavior reproduces for all clients. I tried with the `account` client and got the same "We are sorry... Page not found" in the 2nd tab. I was redirected to URL https://id-staging.avenza.com/realms/avenza/protocol/openid-connect/7315f16d-2315-472b-b8a2-337f06ba4453
and `7315f16d-2315-472b-b8a2-337f06ba4453` is not a `udid` for `account` client. 

When I tried one more time, the URL was different (https://id-staging.avenza.com/realms/avenza/protocol/openid-connect/33341754-17d6-41ca-a138-fdc069ca83bc). It the first tab, where I was able to login successfully, I found this `udid` and it's a `session_state` in the request:
```
https://id-staging.avenza.com/realms/avenza/account/?state=f1ed8fac-92be-4c0c-a5a7-e7ba7766a35f&session_state=33341754-17d6-41ca-a138-fdc069ca83bc&iss=https%3A%2F%2Fid-staging.avenza.com%2Frealms%2Favenza&code={code}
Request Method:
GET
Status Code:
200 OK
```
